server/src/main/java/org/opensearch/action/search/SearchPhase.java

```diff
@@ -34,14 +34,15 @@
 import org.opensearch.common.CheckedRunnable;
 
 import java.io.IOException;
+import java.util.Locale;
 import java.util.Objects;
 
 /**
  * Base class for all individual search phases like collecting distributed frequencies, fetching documents, querying shards.
  *
  * @opensearch.internal
  */
-abstract class SearchPhase implements CheckedRunnable<IOException> {
+public abstract class SearchPhase implements CheckedRunnable<IOException> {
```

Comment:
It seems like the only reason open up `SearchPhase` is to access `SearchPhaseName`, may be it we should m make the enum top level instead?

Reply:
> Also, putting as part of this SearchPhase class make sure that we are limiting the scope

I see it the other way actually - by opening  up the package private class we inflate the scope

Reply:
Let me just move out the enum class, in a separate file, as per your older suggestion

Reply:
thank you!